File: resources/javascript/if-statements.js
let x, b, y, z, yz, myList;

if (x > b) {  // complexity +1
    if ((x > 100 && x <= 122) || !x) {    // complexity +3
        x = 2;
    } else if (x > 122) {    // complexity +1
        x = 300;
    } else {
        x = 1;
    }
    x = 1;
}

x = x > 100 ? 101 : 99;    // complexity +1

/*
    if (not) {
    else if (good) {
        echo "won't be counted! yippi!";
    }
 */

x = x + y;

z = x > 0 ? x : false;    // complexity +1

File: resources/c-sharp/if-statements.cs
int x, y;
double z;

if (x > y) {

    if ( x > 100 || x < 0) {
        y = 1000;
    } else if ( y % x == 1 ) {
        y = 1001;
    } else {
        y = x;
    }
} else {
    x = y;
}

/*
    if (not) {
    else if (good) {
        echo "won't be counted! yippi!";
    }
*/

bool myBool = true;


if (myBool is true or false) {
    Console.Write("hello");
}

myBool = (x is 1); // this should not increase complexity
myBool = (x is 1 and 5); // this should

int? x = (x > 100) ? 12 : null;

string Answer = Answer1 ?? Answer2 ?? Answer3 ?? Answer4;
List<int>? numbers = null;
(numbers ??= new List<int>()).Add(5);
